File: day2/day2.py
with open("input.txt", "r") as f:
    xy = []
    z = []
    for cmd in f.readlines():
        if cmd.startswith("forward"):
            xy.append(int(cmd.replace("forward ", "")))
        elif cmd.startswith("down"):
            z.append(int(cmd.replace("down ", "")))
        elif cmd.startswith("up"):
            z.append(int(cmd.replace("up ", "-")))

print(sum(xy) * sum(z))

aim = 0
pos = 0
depth = 0
with open("input.txt", "r") as f:
    xy = []
    z = []
    for cmd in f.readlines():
        if cmd.startswith("forward"):
            f = int(cmd.replace("forward", ""))
            pos += f
            depth += f * aim
        elif cmd.startswith("down"):
            aim += int(cmd.replace("down ", ""))
        elif cmd.startswith("up"):
            aim += int(cmd.replace("up ", "-"))
print(pos * depth)

File: day2/day2.clj
(defn read_input [file] 
  (def in (slurp file))
  (def splitted (clojure.string/split-lines in))
  (map (fn [line] (clojure.string/split line #" ")) splitted)
)

(defn xy_command [command]
  (case (nth command 0)
    "forward" (read-string(nth command 1))
    "down" 0
    "up" 0
  )
)

(defn z_command [command]
  (case (nth command 0)
    "forward" 0
    "down" (read-string(nth command 1))
    "up" (read-string(format "-%s" (nth command 1)))
  )
)

(def commands (read_input "input.txt"))
(def xy (reduce + (map xy_command commands)))
(def z (reduce  + (map z_command commands)))
(println (* xy z))

(def commands (read_input "input.txt"))

(def aim 
  (reductions + (map z_command commands))
)

(def xy
  (reduce + (map xy_command commands))
)

(def z
  (reduce + (map * aim (map xy_command commands)))
)
(println (* xy z))
